assets/js/stripe.js

```diff
@@ -791,31 +816,38 @@ jQuery( function( $ ) {
 		/**
 		 * Queries the server for a PaymentIntent.
 		 *
+		 * @param {string} savedCard The ID of the saved card. (Optional)
 		 * @return {Promise} A promise that resolves to a payment intent.
 		 */
-		getIntent: function() {
-			return new Promise( function( resolve, reject ) {
-				if ( wc_stripe_form.paymentIntent ) {
-					return resolve( wc_stripe_form.paymentIntent );
-				}
+		getIntent: function( savedCard ) {
+			savedCard = savedCard || 'new';
+
+			// If we have an intent for this card already, use it.
+			if ( wc_stripe_form.paymentIntentCache.hasOwnProperty( savedCard ) ) {
+				return Promise.resolve( wc_stripe_form.paymentIntentCache[ savedCard ] );
```

Comment:
Nitpick: What do you think about putting this condition inside the main `return new Promise( function( resolve ) { ... HERE ... } )`, instead of having multiple points in the function creating Promises? I don't have a strong opinion, whatever oyu think reads best.

Reply:
Generally I always prefer the solution with the least tabs, but in this case I do not mind moving `Promise.resolve` within `new Promise`, so I did it.